### PR TITLE
docs: add workflow badges to the README

### PR DESCRIPTION
**Pull request by tonyandrewmeyer**
Add badges for the Github workflows (except publish) to the README.

These will show when viewing the README in a markdown reader that loads external images (e.g. on Github itself) but more importantly also show the CI status on [the Juju releases page](https://releases.juju.is).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # The `ops` library
 
+![Charmcraft pack](https://github.com/canonical/operator/actions/workflows/charmcraft-pack.yaml/badge.svg)
+![Compatibility: DB charms](https://github.com/canonical/operator/actions/workflows/db-charm-tests.yaml/badge.svg)
+![Compatibility: Basic charms](https://github.com/canonical/operator/actions/workflows/hello-charm-tests.yaml/badge.svg)
+![Compatibility: Observability charms](https://github.com/canonical/operator/actions/workflows/observability-charm-tests.yaml/badge.svg)
+![CI Status](https://github.com/canonical/operator/actions/workflows/framework-tests.yaml/badge.svg)
```
Comment (benhoyt) [inline README.md:7]: Hmm, I'm wondering if we really need/want all of these. Maybe just the Charmcraft pack and CI Status ones?

Comment (tonyandrewmeyer) [reply README.md:7]: I don't think they are super useful in the README itself. I'm not sure what the aim is with the releases page. @jnsgruk what workflows would you like to see showing up there?

Comment (jnsgruk) [reply README.md:7]: I think probably just the `ops test` workflow and the `publish` workflow?

Comment (tonyandrewmeyer) [reply README.md:3]: Thanks, changed!

**Review by jnsgruk — APPROVED** [review]

**Pull request merged**